### DISCUSSION TITLE
How to pass File to a NodeJS server

### DISCUSSION THREAD

**Discussion by jcjpsa** [react-dropzone/react-dropzone#1287] [Q&A]
I already have `react-dropzone` up and running and getting the `File` properly ~~however when I pass it to our NodeJS server for uploading it gets transformed to an object with only the `path` property and value remaining for example `{ path: sample.png }` is sent to the NodeJS server and when I try to upload it is saying the type is incorrect. I use a validation library to check if `File` is passed using `yup`.~~

I temporarily removed the validation but it still behaving the same, I use postman to facilitate the upload and it went perfectly. I think the issue is the `path` is not full and as per research browser will not give you the full path, what is the alternative if I want to send the `File` from `react-dropzone` to NodeJS  server to facilitate the uploading?

**Comment by rolandjitsu** [comment]
@jcjpsa your best bet is the https://developer.mozilla.org/en-US/docs/Web/API/FileReader. Checkout https://react-dropzone.js.org/#usage.
